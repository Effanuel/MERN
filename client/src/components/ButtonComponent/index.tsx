import React from "react";

import "./styles.css";

const ButtonComponent = ({
  label,
  disabled,
  onClick,
  type,
  ...otherProps
}: any) => {
  return (
    <button
      className={type === "close" ? "btn-style btn-cl" : "btn-style btn-su"}
      onClick={onClick}
      disabled={disabled}
      {...otherProps}
    >
      {label}
    </button>
  );
};

export { ButtonComponent };
