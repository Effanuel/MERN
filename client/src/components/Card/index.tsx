import React from "react";

import { Button } from "react-bootstrap";
import "./styles.css";

interface Props {
  name: string;
  login: string;
  password: string;
  onClickUpdate: (
    e: any,
    name: string,
    login: string,
    password: string
  ) => void;
  onClickRemove: (e: any, name: string) => void;
  onClickCopy: (e: any, password: string) => any;
}

const Card = ({
  name,
  login,
  password,
  onClickUpdate,
  onClickRemove,
  onClickCopy
}: Props) => {
  console.log(name, login, password, "card");

  return (
    <div className="container">
      <span
        className="name-login-container"
        onClick={(e: any) => onClickCopy(e, password)}
      >
        <span className="name-style">{name} </span>
        <span className="login-style">{login}</span>
      </span>
      <div className="button-container">
        <Button
          variant="info"
          onClick={(e: any) => onClickUpdate(e, name, login, password)}
        >
          /
        </Button>
        <Button variant="danger" onClick={(e: any) => onClickRemove(e, name)}>
          X
        </Button>
      </div>
    </div>
  );
};

export { Card };
