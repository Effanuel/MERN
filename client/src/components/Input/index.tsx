import React from "react";

import { FormGroup, Form } from "react-bootstrap";

import "./styles.css";

type Props = {
  id?: string;
  type?: any;
  label?: string;
  placeholder: string;
  value?: string;
  onChange: (event: any) => void;
};

const Input = ({
  id,
  type = "text",
  label,
  placeholder,
  value,
  onChange
}: Props) => {
  return (
    <FormGroup style={{ paddingBottom: "0px", marginBottom: "0px" }}>
      {label && <Form.Label>{label}</Form.Label>}
      <Form.Control
        id={id}
        type={type}
        className="input"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        required
      />
    </FormGroup>
  );
};

export { Input };
