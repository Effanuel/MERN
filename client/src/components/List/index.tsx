import React from "react";

import { SearchBar } from "../";

interface Props {
  items: any;
  onClick: any;
  delete?: any;
}
interface State {
  filtered: [object];
  items: [object];
}

const initialState = Object.freeze({
  filtered: [],
  items: []
});

const handleChange = Symbol();

class List extends React.Component<Props, State> {
  readonly state: any = initialState;

  componentDidMount() {
    this.setState({
      items: this.props.items,
      filtered: this.props.items
    });
    console.log(this.props.items);
  }

  static getDerivedStateFromProps(nextProps: any, prevState: any) {
    console.log("DERIVED", nextProps.items !== prevState.items);
    if (nextProps.items !== prevState.items) {
      return { items: nextProps.items };
    } else return null;
  }

  componentDidUpdate(prevProps: any, prevState: any) {
    console.log(prevProps, "=================PROPS PREV");
    console.log(prevState, "=================PROPS STATE");
  }

  // componentDidUpdate(prevProps: any, prevState: any) {
  //   if (prevProps.items !== this.props.items) {
  //     //Perform some operation here
  //     this.setState({ items: someValue });
  //     this.classMethod();
  //   }
  // }

  [handleChange] = (e: any) => {
    // Variable to hold the original version of the list
    let currentList = [];
    // Variable to hold the filtered list before putting into state
    let newList = [];

    // If the search bar isn't empty
    if (e.target.value !== "") {
      // Assign the original list to currentList
      currentList = this.state.items;

      // Use .filter() to determine which items should be displayed
      // based on the search terms
      newList = currentList.filter((item: any) => {
        // change current item to lowercase
        const lc = JSON.stringify(item.name).toLowerCase();
        // change search term to lowercase
        const filter = e.target.value.toLowerCase();
        // check to see if the current list item includes the search term
        // If it does, it will be added to newList. Using lowercase eliminates
        // issues with capitalization in search terms and search content
        return lc.includes(filter);
      });
    } else {
      // If the search bar is empty, set newList to original task list
      newList = this.props.items;
    }
    // Set the filtered state based on what our rules added to newList
    this.setState({
      filtered: newList
    });
  };

  render() {
    return (
      <div>
        <SearchBar
          onChange={this[handleChange]}
          onClick={this.props.onClick}
          placeholder="Search..."
        />
        <ul>
          {this.state.filtered.map((item: any, i: any) => (
            <li key={i}>
              {item.name} &nbsp;
              {item.login} &nbsp;
              {item.password} &nbsp;
              <span
                className="delete"
                onClick={() => this.props.delete(item)}
              />
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

export { List };
