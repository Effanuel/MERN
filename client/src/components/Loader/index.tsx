import React from "react";
import { BarLoader } from "react-spinners";

import styles from "./styles.module.css";

const Loader = ({ loading, height, width, ...other }: any) => {
  return (
    <div className={styles["loader-main"]}>
      <BarLoader loading={loading} height={height} width={"1vw"} {...other} />
    </div>
  );
};

export { Loader };
