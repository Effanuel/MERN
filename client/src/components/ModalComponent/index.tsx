import React from "react";

import { Button, Modal, Form, ProgressBar } from "react-bootstrap";

import "./styles.css";

import { styleProgress } from "../../utils/functions";

interface Props {
  title: string;
  show: boolean;
  onSave: () => void;
  onClose: () => void;
  p_name: string;
  p_login: string;
  p_password: string;
  onInputChange: (event: any) => void;
  onPasswordChange?: (event?: any) => void;
  loadingComponent: any;
  disabled?: boolean;
  progress?: any;
}

const ModalComponent = ({
  title,
  show,
  onSave,
  onClose,
  p_name,
  p_login,
  p_password,
  onInputChange,
  onPasswordChange,
  loadingComponent,
  disabled,
  progress
}: Props) => {
  return (
    <>
      <Modal show={show} onHide={onClose}>
        <Modal.Header closeButton>
          <Modal.Title>{title}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form>
            <Form.Group>
              <Form.Label>Name</Form.Label>
              <Form.Control
                id="name"
                placeholder={p_name}
                onChange={onInputChange}
              />
            </Form.Group>
            <Form.Group>
              <Form.Label>Login</Form.Label>
              <Form.Control
                id="login"
                placeholder={p_login}
                onChange={onInputChange}
              />
            </Form.Group>
            <Form.Group>
              <Form.Label>Password</Form.Label>
              <h1>Generate password</h1>
              <Form.Control
                id="password"
                placeholder={p_password}
                onChange={onPasswordChange}
                type="password"
              />
              <ProgressBar
                variant={styleProgress(progress)}
                now={progress}
                style={{ height: "5px" }}
              />
            </Form.Group>
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={onClose}>
            Close
          </Button>
          <Button variant="primary" onClick={onSave} disabled={disabled}>
            {loadingComponent || "Save Changes"}
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export { ModalComponent };
