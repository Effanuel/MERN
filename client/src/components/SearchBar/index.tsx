import React from "react";

import { Form, InputGroup } from "react-bootstrap";

interface Props {
  onClick: () => void;
  onChange: (event: any) => void;
  placeholder: string;
  value?: any;
}

const SearchBar = ({ onClick, onChange, placeholder, value }: Props) => {
  return (
    <>
      <InputGroup style={{ paddingBottom: "5px" }}>
        <Form.Control
          type="text"
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          required
        />
        <InputGroup.Prepend onClick={onClick}>
          <InputGroup.Text>Add</InputGroup.Text>
        </InputGroup.Prepend>
      </InputGroup>
    </>
  );
};

export { SearchBar };
