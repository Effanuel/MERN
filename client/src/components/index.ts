export * from "./ModalComponent";
export * from "./SpinnerComponent";
export * from "./Card";
