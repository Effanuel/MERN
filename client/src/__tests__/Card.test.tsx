import React from "react";
import { shallow } from "enzyme";
import { SpinnerComponent } from "../components";

describe("SpinnerComponent", () => {
  it("renders component", () => {
    const link = shallow(<SpinnerComponent />);
    expect(link).toMatchSnapshot();
  });
});
