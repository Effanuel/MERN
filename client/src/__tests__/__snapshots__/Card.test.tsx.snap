// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerComponent renders component 1`] = `ShallowWrapper {}`;
